### PR TITLE
Merge 2.1 into 2.2 july 11 2019

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

2.0/2.1 was not merged into 2.2

### How did I fix it?

by merging and back porting changes that did not directly generate a diff due to huge difference between branches

### How to test it?

- Build should be green
- Check resolved conflicts 
```Auto-merging framework/test/mocha/unit/modules/http_api/controllers/node.js
Auto-merging framework/test/mocha/unit/modules/chain/transport/broadcaster.js
CONFLICT (modify/delete): framework/test/mocha/unit/modules/chain/submodules/loader.js deleted in HEAD and modified in release/2.1.0. Version release/2.1.0 of framework/test/mocha/unit/modules/chain/submodules/loader.js left in tree.
Auto-merging framework/test/mocha/unit/modules/chain/peers.js
CONFLICT (content): Merge conflict in framework/test/mocha/unit/modules/chain/peers.js
Auto-merging framework/test/mocha/integration/transactions/5_5_dapps.js
Auto-merging framework/test/mocha/integration/transactions/4_multisignature/4_X_multisignature_edge_cases.js
Auto-merging framework/test/mocha/integration/transactions/0.0.address_collision.js
CONFLICT (content): Merge conflict in framework/test/mocha/integration/transactions/0.0.address_collision.js
Auto-merging framework/test/mocha/integration/state_store/account_store.js
Auto-merging framework/test/mocha/integration/matcher.js
Auto-merging framework/test/mocha/functional/ws/transport/transport.js
CONFLICT (content): Merge conflict in framework/test/mocha/functional/ws/transport/transport.js
Auto-merging framework/test/mocha/functional/http/get/accounts/accounts.js
Auto-merging framework/src/modules/network/index.js
Auto-merging framework/src/modules/http_api/controllers/node.js
CONFLICT (content): Merge conflict in framework/src/modules/http_api/controllers/node.js
CONFLICT (modify/delete): framework/src/modules/chain/submodules/peers.js deleted in HEAD and modified in release/2.1.0. Version release/2.1.0 of framework/src/modules/chain/submodules/peers.js left in tree.
CONFLICT (modify/delete): framework/src/modules/chain/submodules/loader.js deleted in HEAD and modified in release/2.1.0. Version release/2.1.0 of framework/src/modules/chain/submodules/loader.js left in tree.
CONFLICT (modify/delete): framework/src/modules/chain/submodules/blocks/chain.js deleted in HEAD and modified in release/2.1.0. Version release/2.1.0 of framework/src/modules/chain/submodules/blocks/chain.js left in tree.
Auto-merging framework/src/controller/application.js
Auto-merging framework/package.json
Auto-merging framework/package-lock.json
Auto-merging elements/lisk-transactions/test/base_transaction.ts
Auto-merging elements/lisk-transactions/src/base_transaction.ts
CONFLICT (content): Merge conflict in elements/lisk-transactions/src/base_transaction.ts
Automatic merge failed; fix conflicts and then commit the result.
```

### Review checklist

* The PR resolves #INSERT_ISSUE_NUMBER
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/peers.js [hunk 1]
```diff
@@ -47,18 +46,18 @@ class Peers {
...
-			await this.channel.invoke('network:getPeersCountByFilter', {
-				state: PEER_STATE_CONNECTED,
-			}),
+			this.channel.invoke('network:getConnectedPeersCountByFilter', {}),
 			MAX_PEERS
 		);
 
 		const matchedCount = Math.min(
-			await this.channel.invoke('network:getPeersCountByFilter', {
+			this.channel.invoke('network:getConnectedPeersCountByFilter', {
```
Comment (shuse2) [inline framework/src/modules/chain/peers.js]: we need `await` here, although it's not a problem now

File: framework/src/modules/chain/peers.js [hunk 2]
```diff
@@ -47,18 +46,18 @@ class Peers {
...
-	async calculateConsensus(broadhash) {
+	// eslint-disable-next-line class-methods-use-this
+	async calculateConsensus() {
+		const { broadhash } = await this.channel.invoke('app:getApplicationState');
+
 		const activeCount = Math.min(
-			await this.channel.invoke('network:getPeersCountByFilter', {
-				state: PEER_STATE_CONNECTED,
-			}),
+			this.channel.invoke('network:getConnectedPeersCountByFilter', {}),
```
Comment (shuse2) [inline framework/src/modules/chain/peers.js]: we need `await` here, although it's not a problem now

File: elements/lisk-transactions/src/0_transfer_transaction.ts [hunk 3]
```diff
@@ -104,18 +101,6 @@ export class TransferTransaction extends BaseTransaction {
 			);
 		}
 
-		if (!this.fee.eq(TRANSFER_FEE)) {
```
Comment (SargeKhan) [inline elements/lisk-transactions/src/0_transfer_transaction.ts]: why we remove this constraint from this place?

File: elements/lisk-transactions/src/4_multisignature_transaction.ts [hunk 4]
```diff
@@ -228,6 +210,22 @@ export class MultisignatureTransaction extends BaseTransaction {
 		return errors;
 	}
 
+	public validateFee(): TransactionError | undefined {
+		const expectedFee = new BigNum(MultisignatureTransaction.FEE).mul(
```
Comment (SargeKhan) [inline elements/lisk-transactions/src/4_multisignature_transaction.ts]: Multisig transaction has a constant fee irrespective of keys group length. Did we update the fee calculation already?

File: elements/lisk-transactions/src/0_transfer_transaction.ts (hunk 3 above)
Comment (shuse2) [reply elements/lisk-transactions/src/0_transfer_transaction.ts]: it's checked in the Bese transaction

File: elements/lisk-transactions/src/4_multisignature_transaction.ts (hunk 4 above)
Comment (shuse2) [reply elements/lisk-transactions/src/4_multisignature_transaction.ts]: I think the calculation is correct per current protocol for register multi-signature transaction

Comment (SargeKhan) [reply elements/lisk-transactions/src/4_multisignature_transaction.ts]: My bad. Looked at the protocol definition again.

**Pull request merged**